src/imfilter.jl

```diff
@@ -6,18 +6,31 @@ function imfilter_cpu(img, krn)
   imfilter(img, centered(krn), Inner(), Algorithm.FFT())
 end
 
+# preallocated memory for krn optimization on GPU
+krnbuffer_gpu = nothing
+
+function alloc_krn_buffer_gpu!(sz)
+  global krnbuffer_gpu
+  
+  if isnothing(krnbuffer_gpu) || size(krnbuffer_gpu) != sz
+    krnbuffer_gpu = CuArray(zeros(sz))
+  end
+
+  krnbuffer_gpu
+end
+
 function imfilter_gpu(img, krn)
   # retrieve basic info
   N = ndims(img)
   T = eltype(img)
 
-  # pad kernel to common size with image
-  padsize = size(img) .- size(krn)
-  padkrn  = padarray(krn, Fill(zero(T), ntuple(i->0, N), padsize))
+  # copy krn data to preallocated GPU memory
+  paddedkrn = alloc_krn_buffer_gpu!(size(img))
+  copyto!(paddedkrn, CartesianIndices(krn), CuArray(krn), CartesianIndices(krn))
 
   # perform ifft(fft(img) .* conj.(fft(krn)))
   fftimg = img |> CUFFT.fft
-  fftkrn = padkrn |> CuArray |> CUFFT.fft
+  fftkrn = paddedkrn |> CUFFT.fft
```

Comment:
Please try to not rename variables unless strictly needed. padkrn seems like a perfectly valid name. You should just erase CuArray in this line.

Reply:
ok, its fixed